api/api.go

```diff
@@ -736,22 +750,24 @@ func (f *PublicController) GetMaterialFromCourse(c *gin.Context) {
 	user_id, err := f.GetIdFromCookie(c)
 	if err != nil {
 		log.Errorf("Unable to get id from Cookie: %s\n", err.Error())
-		c.Status(http.StatusInternalServerError)
+		c.Status(http.StatusBadRequest)
 		return
 	}
 	course_id, err := strconv.Atoi(c.Param("id"))
 	if err != nil {
-		log.Errorf("Unable to convert parameter `id` to int: %s", err.Error())
-		c.Status(http.StatusInternalServerError)
+		log.Errorf("Unable to convert parameter `id` to string: %s\n", err.Error())
```

Comment:
Es wird von einem `string` zu einem `int` konvertiert. `Atoi` entspricht "Ascii-to-integer".
Bitte überall abändern.